modules/account-lib/src/coin/dot/transaction.ts

```diff
@@ -175,6 +182,61 @@ export class Transaction extends BaseTransaction {
     return result;
   }
 
+  explainTransaction(): TransactionExplanation {
+    const result = this.toJson();
+    const displayOrder = ['outputAmount', 'changeAmount', 'outputs', 'changeOutputs', 'fee', 'type'];
+    const outputs: TransactionRecipient[] = [];
+    const explanationResult: TransactionExplanation = {
+      displayOrder,
+      outputAmount: result.amount?.toString() || '0',
+      changeAmount: '0',
+      changeOutputs: [],
+      outputs,
+      fee: {
+        fee: result.tip?.toString() || '',
+        type: 'tip',
+      },
+      type: this.type,
+    };
+
+    if (this.type === TransactionType.Send) {
+      explanationResult.outputs = [
+        {
+          address: result.to?.toString() || '',
+          amount: result.amount?.toString() || '',
+        },
+      ];
+      explanationResult.owner = result.owner;
+      explanationResult.forceProxyType = result.forceProxyType;
+      explanationResult.displayOrder.push('owner', 'forceProxyType');
+    }
+    if (this.type === TransactionType.StakingActivate) {
+      explanationResult.outputs = [
+        {
+          address: result.controller?.toString() || '',
+          amount: result.amount || '',
+        },
+      ];
+      explanationResult.payee = result.payee;
+      explanationResult.displayOrder.push('payee');
+    }
+    if (this.type === TransactionType.WalletInitialization) {
+      explanationResult.owner = result.owner;
+      explanationResult.proxyType = result.proxyType;
+      explanationResult.delay = result.delay;
+      explanationResult.displayOrder.push('owner', 'proxyType', 'delay');
+    }
+    if (this.type === TransactionType.StakingUnlock) {
+      explanationResult.outputs = [
+        {
+          address: result.sender.toString(),
+          amount: result.amount || '',
+        },
+      ];
+    }
```

Comment:
This should be a `switch/case`. Also, each `case` body should be its own method

Reply:
Also on the switch the default value should have an error message, mentioning that we not support this type of tx. Because we have many type of tx types in polka with all the existent extrinsic.